homeassistant/components/nws/weather.py

```diff
@@ -102,11 +113,10 @@ def __init__(self, nws, mode, units):
     async def async_added_to_hass(self) -> None:
         """Set up a listener and load data."""
         self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass,
-                signal_unique_id(self.latitude, self.longitude),
-                self._update_callback,
-            )
+            self.coordinator_observation.async_add_listener(self._update_callback)
+        )
+        self.async_on_remove(
+            self.coordinator_forecast.async_add_listener(self._update_callback)
         )
         self._update_callback()
 
```

Comment:
If a we implement the entity method `async_update` and await `coordinator.async_request_refresh` inside we allow the users to schedule manual updates using our general entity update service.